source/parser.hera

```diff
@@ -7989,22 +7990,34 @@ Init
             return
           }
 
-          let { pattern, block } = c
+          let { patterns, block } = c
+          let pattern = patterns[0]
 
           const indent = block.expressions?.[0]?.[0] || ""
 
           // TODO: intermediate refs and ref decs
+          // TODO: multiple binding patterns
 
-          const conditions = []
-          getPatternConditions(pattern, ref, conditions)
+          const alternativeConditions = patterns.map((pattern, i) => {
+            const conditions = []
+            getPatternConditions(pattern, ref, conditions)
+            return conditions
+          })
 
-          const condition = {
-            type: "ParenthesizedExpression",
-            children: ["(", conditions.map((c, i) => {
+          const conditionExpression = alternativeConditions.map((conditions, i) => {
+            const conditionArray = conditions.map((c, i) => {
               if (i === 0) return c
               return [" && ", ...c]
-            }), ")"],
-            expression: conditions,
+            })
+
+            if (i === 0) return conditionArray
+            return [" || ", ...conditionArray]
+          })
+
+          const condition = {
```

Comment:
We could use `module.makeLeftHandSideExpression` here, which avoids parens in some simple cases. Though maybe we need a variety of such parenthesization functions...

Reply:
Oops, sorry, makes sense. We could probably still make a "parenthesize" helper, but we're building nodes all over the place so not sure that's worth singling out.